test/python/circuit/test_parameters.py

```diff
@@ -346,6 +346,23 @@ def test_multiple_named_parameters(self):
         self.assertEqual(theta.name, "θ")
         self.assertEqual(qc.parameters, {theta, x})
 
+    @named_data(
+        ["int", 2, int],
+        ["float", 2.1, float],
+        ["float16", numpy.float16(2.1), float],
+        ["float32", numpy.float32(2.1), float],
+        ["float64", numpy.float64(2.1), float],
```

Comment:
I saw float16 and float32 tested in test_bind_half_single_precision so I tested them here. We could test other types if any are relevant.